### ISSUE TITLE
Babel compiler output contains invalid filename and sourceFile paths when testing local packages

### ISSUE THREAD

**Issue by jankapunkt** [meteor/meteor#13287]
## Background

I currently update the Meteor-Coverage package and use `babel-plugin-istanbul` to instrument the files for test-coverage.
This works fine for testing apps but fails for package tests (reproduction steps below).

## Problem

The `babel-plugin-istanbul` seems to receive wrong filename / sourceFile paths when testing local packages. Istanbul can instrument the files and the coverage summary is correctly computed:

![image](https://github.com/user-attachments/assets/44525605-0c73-4213-944e-8567a4da323d)

However, the source files are not reachable due to a wrong path (clicking on  `packages/lmieulet:meteor-coverage/server/boot.js`):

<details>
<summary>Error on instrumenter detail for meteor-coverage/server/boot.js</summary>

 ```
Unable to lookup source: /home/jankapunkt/dev/leaonline/lib/meteor-coverage/someapp/packages/lmieulet:meteor-coverage/server/boot.js(ENOENT: no such file or directory, open '/home/jankapunkt/dev/leaonline/lib/meteor-coverage/someapp/packages/lmieulet:meteor-coverage/server/boot.js')
Error: Unable to lookup source: /home/jankapunkt/dev/leaonline/lib/meteor-coverage/someapp/packages/lmieulet:meteor-coverage/server/boot.js(ENOENT: no such file or directory, open '/home/jankapunkt/dev/leaonline/lib/meteor-coverage/someapp/packages/lmieulet:meteor-coverage/server/boot.js')
    at Context.defaultSourceLookup [as sourceFinder] (/home/jankapunkt/dev/leaonline/lib/meteor-coverage/.npm/package/node_modules/istanbul-lib-report/lib/context.js:15:15)
    at Context.getSource (/home/jankapunkt/dev/leaonline/lib/meteor-coverage/.npm/package/node_modules/istanbul-lib-report/lib/context.js:78:17)
    at annotateSourceCode (/home/jankapunkt/dev/leaonline/lib/meteor-coverage/.npm/package/node_modules/istanbul-reports/lib/html/annotator.js:217:40)
    at HtmlReport.onDetail (/home/jankapunkt/dev/leaonline/lib/meteor-coverage/.npm/package/node_modules/istanbul-reports/lib/html/index.js:401:33)
    at packages/lmieulet:meteor-coverage/server/report/report-html.js:85:14
    at Array.forEach ()
    at Object.generate (packages/lmieulet:meteor-coverage/server/report/report-html.js:80:14)
    at Object.generateReport (packages/lmieulet:meteor-coverage/server/report/report-service.js:73:14)
    at Object.exportFile (packages/lmieulet:meteor-coverage/server/handlers.js:86:19)
    at packages/lmieulet:meteor-coverage/server/router.js:53:16
    at packages/lmieulet:meteor-coverage/server/router.js:26:19
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
```

</details>

The cause seems to be very deep, since this wrong path is already used by the babel plugin, which basically just injects the `filename` value into the transformed code. As an example I show an excerpt from the compiled package file

```shell
/tmp/meteor-test-run49v8le.bwwnr/.meteor/local/build/programs/server/packages/local-test_lmieulet_meteor-coverage.js` 
```

There you can see istanbul injected the wrong path directly:

<details>
<summary>wrong injected path in instrumented file</summary>

```js
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

},"boot.js":function module(require,exports,module){

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//                                                                                                                    //
// packages/lmieulet_meteor-coverage/server/boot.js                                                                   //
//                                                                                                                    //
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
                                                                                                                      //
!module.wrapAsync(async function (module, __reifyWaitForDeps__, __reify_async_result__) {
  "use strict";
  try {
    let fs;
    module.link("node:fs", {
      default(v) {
        fs = v;
      }
    }, 0);
    let path;
    module.link("node:path", {
      default(v) {
        path = v;
      }
    }, 1);
    let mkdirp;
    module.link("mkdirp", {
      default(v) {
        mkdirp = v;
      }
    }, 2);
    let COVERAGE_APP_FOLDER, COVERAGE_EXPORT_FOLDER;
    module.link("./context/conf", {
      COVERAGE_APP_FOLDER(v) {
        COVERAGE_APP_FOLDER = v;
      },
      COVERAGE_EXPORT_FOLDER(v) {
        COVERAGE_EXPORT_FOLDER = v;
      }
    }, 3);
    let Log;
    module.link("./context/log", {
      default(v) {
        Log = v;
      }
    }, 4);
    let Router;
    module.link("./router", {
      default(v) {
        Router = v;
      }
    }, 5);
    if (__reifyWaitForDeps__()) (await __reifyWaitForDeps__())();
    function cov_2ffb0vaczy() {
      var path = "/home/jankapunkt/dev/leaonline/lib/meteor-coverage/someapp/packages/lmieulet:meteor-coverage/server/boot.js";
      var hash = "c6995ace23eb22ba41a85116d5f1e139cca5fde7";
      var global = new Function("return this")();
      var gcv = "__coverage__";
      var coverageData = {
        path: "/home/jankapunkt/dev/leaonline/lib/meteor-coverage/someapp/packages/lmieulet:meteor-coverage/server/boot.js",
        statementMap: {
          "0": {
            start: {
              line: 8,
              column: 13
            },
            end: {
              line: 26,
              column: 1
            }
          },
          "1": {
            start: {
```

</details>

## Expected behaviour (as it was in Meteor 2)

Pass correct paths to babel plugins

## Reproduction

TL;DR: there is also a repo here https://github.com/jankapunkt/babel-compiler-bug-repro

```shell
$ meteor create --release 3.0.1 host-app
$ mkdir -p packages
$ meteor create --package foo:bar
$ cd bar
$ echo '{"env":{"COVERAGE":{"plugins":["istanbul"]}}}' >> .babelrc
$ cd ../../
$ BABEL_ENV=COVERAGE meteor test-packages ./packages/bar
```

Got to the `/tmp/meteor-test-run<hash>` test project and go for `.meteor/local/build/programs/server/packages/foo_bar.js` and see the `path` being 

`/home/jankapunkt/dev/leaonline/babel-compiler-bug-repro/host-app/packages/foo:bar/bar.js` 

instead of

`/home/jankapunkt/dev/leaonline/babel-compiler-bug-repro/host-app/packages/foo_bar/bar.js`

## What I did

I searched the git history of various packages and tools (babel-compiler, meteor-babel, isobuild, package catalog) but I could not find the commit that causes this issue (since it worked in Meteor 2).

**Comment by jankapunkt** [comment]
Update: if you run the [repro](https://github.com/jankapunkt/babel-compiler-bug-repro) with Meteor 2.16 you'll also already get these paths so it seems to be rather coming from the NPM package than Meteor itself.
